### ISSUE TITLE
Mixing unordered lists with ordered lists

### ISSUE THREAD

**Issue by gonzae** [markedjs/marked#1308]
**Describe the bug**
When I write an unordered list and right after an ordered list, both lists get merged. Not expected behavior. 

**To Reproduce**

Write these example lists:
```
* bullet
* bullet
1. one
2. two
3. three
```
You get this output:
```
<ul>
<li>bullet</li>
<li>bullet</li>
<li>one</li>
<li>two</li>
<li>three</li>
</ul>
```

1. [Marked Demo](https://marked.js.org/demo/?outputType=html&text=*%20bullet%0A*%20bullet%0A1.%20one%0A2.%20two%0A3.%20three)
2. [CommonMark Demo](https://spec.commonmark.org/dingus/?text=*%20bullet%0A*%20bullet%0A1.%20one%0A2.%20two%0A3.%20three%0A)

**Expected behavior**
I would expect to have both lists, an unordered list and right after, an ordered list.

**Comment by Martii** [comment]
If you think that is fun... check these out. Ran into this yesterday with double spacing which I didn't even know existed... or does it really?

GFM demo:

``` md
* bullet
* bullet
* bullet

* bullet
```

* bullet
* bullet
* bullet

* bullet

[Marked Demo](https://marked.js.org/demo/?text=*%20bullet%0A*%20bullet%0A*%20bullet%0A%0A*%20bullet)
[CommonMark Demo](https://spec.commonmark.org/dingus/?text=*%20bullet%0A*%20bullet%0A1.%20one%0A2.%20two%0A3.%20three%0A)

**Comment by UziTech** [comment]
@Martii that is fixed in #1304

**Comment by Martii** [comment]
@UziTech 
Had a suggestion... is it possible on the Marked Demo site to have use current HEAD option as well or is that too much? e.g. the equivalent of "nightly"... just to compare?

**Comment by UziTech** [comment]
I have a branch that will allow you to pick the version, but I haven't had time to work on it.

**Comment by slang800** [comment]
Babelmark demo: https://johnmacfarlane.net/babelmark2/?normalize=1&text=*+bullet%0A*+bullet%0A1.+one%0A2.+two%0A3.+three

Looks like this is a pretty common issue across implementations.

**Comment by styfle** [comment]
@slang800 It looks like this is the correct behavior when `pedantic: true` since that's how the original markdown.pl performs.

@UziTech Does PR #1304 take into account the `pedantic` option?

**Comment by styfle** [comment]
The original issue from @gonzae is **not** resolved in 0.5.0 but the issue from @Martii is resolved in 0.5.0

As @slang800 pointed out with the babelmark demo, we are currently compatible with the original markdown.pl so the output is correct when the option `pedantic: true` but we should follow CommonMark when `pedantic: false`.

**Comment by UziTech** [comment]
looks like this was fixed in v0.6.0
